packages/core-js-sdk/src/sdk/magicLink/index.ts

```diff
@@ -99,11 +101,18 @@ const withMagicLink = (httpClient: HttpClient) => ({
       (acc, delivery) => ({
         ...acc,
         [delivery]: withUpdatePhoneValidations(
-          (loginId: string, phone: string, URI?: string, token?: string) =>
+          (
+            loginId: string,
+            phone: string,
+            URI?: string,
+            token?: string,
+            addToLoginIDs?: boolean,
+            onMergeUseExisting?: boolean
+          ) =>
```

Comment:
We are casting it to `MagicLink[Routes.updatePhone]` so we need to update the type as well

In addition, it seems like we have several optional parameters, 
I would consider putting it in an object for easier usage
e.g. `...phone.sms(loginId, phone, {URI, token, addToLoginIDs, ...})`

In this way, in case we want to pass only the last param `onMergeUseExisting = true` we don't have to call the fn in this way 
`...phone.sms(loginId, phone, undefined, undefined, undefined, true)`